addons/stock_account/models/stock_move.py

```diff
@@ -445,6 +445,11 @@ def _generate_analytic_lines_data(self, unit_amount, amount):
 
     def _generate_valuation_lines_data(self, partner_id, qty, debit_value, credit_value, debit_account_id, credit_account_id, description):
         # This method returns a dictionary to provide an easy extension hook to modify the valuation lines (see purchase for an example)
+        def get_credit_debit_from_value(amount):
+            if self.env.context.get('is_returned') and self.env.company.account_storno:
```

Comment:
We shouldn't do `self.env.company.account_storno`.
If it works, it will be by chance because someone would have done something such as `.with_company(company)` in the calling code.

If we do all the step with company A (storno = False) and we post the invoice with company B (storno = True), then we would get a traceback.
And the reverse: all steps with B storno True and post with A storno False, we don't get the negative credit/debit.

If we don't have the choice, we need to have a test that demonstrates that there isn't any issue posting an invoice from another company that wouldn't have storno.

Reply:
Hello! I understand what you mean but that's not really working "by chance", it's working with `with_company` by design.

https://github.com/odoo/odoo/blob/af7be17b71fc72d15149c4baabadeacdcf77cc99/addons/stock_account/models/stock_move.py#L560-L587

But I see your point, I will add a test reversing a stock move from another company without storno.

Reply:
Can't you use `self.company_id.account_storno` instead ?

Reply:
Hello, no, the account move is not in the same company_id as the stock move (see all the with_company in my previous comment).